hub/config-schema.json

```diff
@@ -1,7 +1,9 @@
 {
     "$schema": "http://json-schema.org/draft-07/schema#",
+    "additionalProperties": false,
```

Comment:
Do we want to be this strict?  What's the downside of allowing anything?  I could see this becoming a source of compatibility problems down the road.

Reply:
None of this is strict because its console warnings. I've found this to actually be the most useful warning after setting up a config with something like `server_name` vs `serverName`.

Reply:
Note: I don't think we'd ever end up making this fully strict because some of the cloud credentials are very free-form

Reply:
Ah, so to be clear -- if I add extra properties and use `Ajv` to validate the config against this schema, this will only result in a warning?  i.e. the program will otherwise continue to run?

Reply:
Yeah that is correct, only outputs to `console.warn`